app/src/main/java/com/github/se/icebreakrr/ui/authentication/ProfileCreation.kt

```diff
@@ -0,0 +1,314 @@
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.github.se.icebreakrr.model.profile.Gender
+import com.github.se.icebreakrr.model.profile.Profile
+import com.github.se.icebreakrr.model.profile.ProfilesViewModel
+import com.github.se.icebreakrr.model.tags.TagsViewModel
+import com.github.se.icebreakrr.ui.navigation.NavigationActions
+import com.github.se.icebreakrr.ui.navigation.TopLevelDestinations
+import com.github.se.icebreakrr.ui.profile.ProfilePictureSelector
+import com.github.se.icebreakrr.ui.tags.TagSelector
+import com.google.firebase.Firebase
+import com.google.firebase.Timestamp
+import com.google.firebase.auth.auth
+import com.google.firebase.messaging.FirebaseMessaging
+import java.util.Calendar
+
+@Composable
+fun ProfileCreationScreen(
+    tagsViewModel: TagsViewModel,
+    profilesViewModel: ProfilesViewModel,
+    navigationActions: NavigationActions
+) {
+
+  val configuration = LocalConfiguration.current
+  val screenWidth = configuration.screenWidthDp.dp
+  val screenHeight = configuration.screenHeightDp.dp
+
+  val profilePictureSize = screenWidth * 0.25f
+
+  val scrollState = rememberScrollState()
+  val fullName = remember { mutableStateOf("") }
+  val catchphrase = remember { mutableStateOf("") }
+  val description = remember { mutableStateOf("") }
+
+  val tempBitmap = profilesViewModel.tempProfilePictureBitmap.collectAsState().value
+  val user = profilesViewModel.selectedProfile.collectAsState().value
+
+  val selectedTags = tagsViewModel.filteringTags.collectAsState().value
+  val tagsSuggestions = tagsViewModel.tagsSuggestions.collectAsState()
+  val stringQuery = tagsViewModel.query.collectAsState()
+
+  val expanded = remember { mutableStateOf(false) }
+
+  var showDialog by remember { mutableStateOf(false) }
+  var isModified by remember { mutableStateOf(false) }
+
+  val context = LocalContext.current
+  val calendar =
+      Calendar.getInstance().apply {
+        set(Calendar.HOUR_OF_DAY, 0)
+        set(Calendar.MINUTE, 0)
+        set(Calendar.SECOND, 0)
+        set(Calendar.MILLISECOND, 0)
+        add(Calendar.YEAR, -20)
+      }
+  val defaultDate = calendar.time
+
+  val birthdate = remember { mutableStateOf(DateUtils.formatDate(defaultDate)) }
+  val selectedDate = remember { mutableStateOf(defaultDate) }
+  var isDateValid = remember { mutableStateOf(true) }
+
+  var selectedGender by remember { mutableStateOf<Gender?>(null) }
+
+  // Add form validation
+  fun isFormValid(): Boolean {
+    return fullName.value.isNotBlank() &&
+        selectedDate.value != null &&
+        isDateValid.value &&
+        catchphrase.value.isNotBlank() &&
+        description.value.isNotBlank() &&
+        selectedGender != null
+  }
+
+  // Create DatePicker
+  val datePicker =
+      DatePickerDialog(
+          context,
+          { _, year, month, dayOfMonth ->
+            val newCalendar =
```

Comment:
You use twice the same instance of calendar
Copy it into one variable and reuse the variable

Reply:
Yes I'll change that!